aptos-move/vm-genesis/src/lib.rs

```diff
@@ -1105,12 +1225,19 @@ fn mainnet_genesis_config() -> GenesisConfiguration {
     }
 }
 
-#[derive(Debug, Clone, Serialize, Deserialize, Hash, PartialEq, Eq)]
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, PartialOrd, Ord)]
 pub struct AccountBalance {
     pub account_address: AccountAddress,
     pub balance: u64,
 }
 
+impl Hash for AccountBalance {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        state.write(&self.account_address.to_vec());
```

Comment:
I see that the balance is not factored in here. Do we ensure that an account only has a single entry in the config?

Reply:
Yep, this is actually to mitigate a flaw in the [genesis tx encoder functions](https://github.com/Entropy-Foundation/aptos-core/blob/dev/aptos-move/vm-genesis/src/lib.rs#L118) of Aptos. We see that they require a slice of `AccountBalance`, but it then become the burden of the caller function to make sure that there aren't duplicate accounts in this slice. So to make sure that only one account exists in such list, we need to hash only the `account_address` in this `AccountBalance` type.